lesson05/assignment/css/style.scss

```diff
@@ -4,3 +4,53 @@
 @import 'settings.responsive';
 @import 'components.content';
 
+
+.container {
+   height: 100vh;
+   width: 100vw;
+   margin: 0 auto;
+   display: grid;
+   grid-template-areas: "header header"
+                        "nav nav"
+                        "main right"
+                        "footer footer";
+   grid-template-rows: 1fr 3rem 3fr 1fr;
+   grid-template-columns: 2fr 1fr;
+}
+.header {
+   background: deepskyblue;
+   grid-area: header;
+}
+.nav {
+   background: pink;
+   grid-area: nav;
+}
+.main {
+   background: #1f567c;
+   grid-area: main;
+}
+.aside {
+   background: gold;
+   grid-area: right;
+ }
+.footer {
+   background: blueviolet;
+   grid-area: footer;
+}
+.cell {
+   color: white;
+   font-size: 2rem;
+   text-align: center;
+   padding: 1rem;
+}
+
+//for viewport width 767px and less /* flex layout */
+@media screen and (max-width: 767px) {
+  .container {
+    display: flex;
```

Comment:
This is the code that is causing some strange behavior. You will still want to use grid. Try experimenting with changing your grid-template-areas to maybe make your weather jokes and weather section stack on smaller screens. See if this article helps explain this: [https://thoughtbot.com/blog/concise-media-queries-with-css-grid](https://thoughtbot.com/blog/concise-media-queries-with-css-grid)

Reply:
Thank you, Tyler, for the great concise-media help link, I was able to fix the issue, the responsive behavior is working correctly now.